ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobMetadata.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.utils.Utils;
+import java.util.Objects;
+
+
+/**
+ * Blob metadata document POJO class.
+ */
+public class CloudBlobMetadata {
+  public static final String FIELD_CREATION_TIME = "creationTime";
+  public static final String FIELD_UPLOAD_TIME = "uploadTime";
+  public static final String FIELD_DELETION_TIME = "deletionTime";
+  public static final String FIELD_EXPIRATION_TIME = "expirationTime";
+  public static final String FIELD_ACCOUNT_ID = "accountId";
+  public static final String FIELD_CONTAINER_ID = "containerId";
+
+  private String id;
+  private String partitionId;
+  private long creationTime;
+  private long uploadTime;
+  private long size;
+  private int accountId;
+  private int containerId;
+  private long expirationTime;
```

Comment:
It's not possible to make any members final/immutable, because it has to be created from a json document using json serialization which requires no-op constructor and setters for all fields.